pretix_payone/signals.py

```diff
@@ -88,3 +88,11 @@ def signal_process_response(sender, request, response, **kwargs):
 
 
 settings_hierarkey.add_default("payment_payone_method_creditcard", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_V", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_M", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_J", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_A", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_D", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_O", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_U", True, bool)
+settings_hierarkey.add_default("payment_payone_cardtypes_P", True, bool)
```

Comment:
Would it make sense to add these dynamically so they are always in sny with `payments.cardtypes`? Like so (not tested):

```suggestion
from .payment import cardtypes
for k, v in cardtypes:
    settings_hierarkey.add_default("payment_payone_cardtypes_{}".format(k), True, bool)
```

Reply:
I was thinking of that too - but decided against it as this is just reinstating the previous state for people already using the plugin where those payment methods were enabled by default.

The only issue I am seeing with your proposed code is: What do we want to happen, if there is a new card payment type added in the future?

For completely new events, we probably don't care: They might as well get the new card type enabled by default.

For existing events on the other hand, we probably do not want a new card type enabled without the consent of the event-organizer.

So one other way might be to not enable any card types by default and add a migration that only populates these card types for events which are already using this plugin.

Any thoughts on this?

Reply:
Ah, good catch, you are absolutely right! I just wanted to avoid repetition of these „magic“ strings, but setting new cardtypes to true for old installations is definitely not what we want. To still avoid duplication of these strings and needing to edit only one place, when adding a new card-type, one could add a „default“-value to the cardtypes-tuple in payment.py like so:

```
# (key, label, activated-by-default)
cardtypes = [
    ("V", _("VISA"), true),
...
```

adding a new cardtype would then just mean adding one line:

   `("N", _("NEW CARD"), false),`

And the loop would change to:

```suggestion
from .payment import cardtypes
for k, v, d in cardtypes:
    settings_hierarkey.add_default("payment_payone_cardtypes_{}".format(k), d, bool)
```

Note though, there are other places we loop through cardtypes, that need to be adjusted slightly. So instead of `for k, v in cardtypes:` we need `for k, v, d in cardtypes:`

What do you think? But explicitly adding `settings_hierarkey` is fine as well.

Reply:
I slept on it and I think we might have an impossible situation to solve here...

Assuming Visa + Mastercard are current card types, either solution (no matter if my wall of code or your new shorthand) will be enabled by default for existing users of pprov and prechecked for new users of the pprov.

Now we add another card, Amex. By using your proposal and setting the bool to `false`, existing users won't get the cardtype enabled without their consent. But new users will be faced with an interesting display: Visa + Mastercard will be prechecked on the form, whereas Amex won't. Which is probably not something we want - new users should probably get all payment methods prechecked it save them a few click.

(Yes, I'm aware that this is an issue with my old code, too)

So perhaps we'll go the other way round: We use your solution and make all card types enabled by default for now (and in the future). But if we ever add a new card method, we inject a "false" for those for existing users?

Reply:
You are absolutely right, that is tricky. I have two ideas:
A) If we want the cardtypes for new events be enabled by default, everytime we add a new cardtype, we need to add a migration to set the new cardtype on exisiting events to `false`. Not really ideal and easy to forget to write the migration. Is that what you mean by 

> inject a "false" for those for existing users?

B) If we turn it the other way around and as default have the cardtypes disabled then we would not need migrations when adding new cardtypes. New events would need a few clicks to select all CardTypes. What about, when showing the SettingsForm, we pre-check/enable all cardtypes if the payment-method is currently disabled and all cardtypes are disabled. So basically for new events we enable all cardtypes just in the view/settingsform when it is first activated.

Reply:
a) Yes, exactly what I meant
b) I also have no problem whatsoever to force people to check all the credit card types upon setting up the payment method. Clean solution, requires no hierarkey-defaults, we can even avoid checking them when first displaying the SettingsField. But then we're back at "we need a migration for this release for existing card users".

Reply:
I vote for default "false" and having users click the boxes. We'll need a migration now, but never again, unlike in the other approaches.